services/file_service.proto

```diff
@@ -30,46 +43,95 @@ import "transaction_response.proto";
+ * restored, so it is important to consider this when assigning keys for
+ * a file.<br/>
+ * If any of the keys in a `KeyList` are complex, the full requirements of
+ * each complex key must be met to count as a "valid signature" for that key.
+ * A complex key structure (i.e. a `ThresholdKey`, or `KeyList`, possibly
+ * including additional `ThresholdKey` or `KeyList` descendants) may be
+ * assigned as the sole entry in a file `keys` field to ensure all transactions
+ * have the same signature requirements.
  */
 service FileService {
     /**
-     * Creates a file
+     * Create a file in HFS.
      */
     rpc createFile (Transaction) returns (TransactionResponse);
 
     /**
-     * Updates a file
+     * Update a file in HFS.
      */
     rpc updateFile (Transaction) returns (TransactionResponse);
 
     /**
-     * Deletes a file
+     * Delete a file in HFS.<br/>
+     * The content of a file deleted in this manner is completely removed
+     * from network state.
      */
     rpc deleteFile (Transaction) returns (TransactionResponse);
 
     /**
-     * Appends to a file
+     * Append content to a file in HFS.
      */
     rpc appendContent (Transaction) returns (TransactionResponse);
 
     /**
-     * Retrieves the file contents
+     * Retrieve the content of a file in HFS.<br/>
+     * Note that this query retrieves _only_ the file content, not any of
+     * the metadata for the file.
      */
     rpc getFileContent (Query) returns (Response);
 
     /**
-     * Retrieves the file information
+     * Retrieve the metadata for a file in HFS.<br/>
+     * Note that this query does not retrieve the file _content_.
      */
     rpc getFileInfo (Query) returns (Response);
 
     /**
-     * Deletes a file if the submitting account has network admin privileges
+     * Delete a "regular" file without "owner" authorization.<br/>
+     * This transaction _does not_ require signatures for the keys in
+     * the file `keys` list, but must be signed by a "privileged" account.
+     * <p>
+     * This transaction SHALL NOT accept a file identifier for
+     * a "system" file.<br/>
+     * This transaction SHALL NOT remove the _content_ of the file from state.
+     * This permits use of the `systemUndelete` to reverse this action if
+     * performed in error.
+     * <p>
+     * This is a privileged transaction, and only accounts 2-59 are permitted
+     * to call this function, by default. The actual restriction is in the
+     * `api-permission.properties` file in the consensus node configuration.
      */
     rpc systemDelete (Transaction) returns (TransactionResponse);
 
     /**
-     * Undeletes a file if the submitting account has network admin privileges
+     * Undelete a "regular" file.
+     * This transaction must be signed by a "privileged" account.<br/>
+     * <p>
+     * This transaction SHALL NOT accept a file identifier for
+     * a "system" file.<br/>
+     * The file identified MUST be deleted.<br/>
```

Comment:
```suggestion
     * The file identified MUST have previously been deleted.<br/>
```

Reply:
Actually I'm not seeing any checks in the code that the file was previously deleted. It looks to me like you could call systemUndelete on a file that was not previously deleted, and it would do nothing but charge you a transaction fee. @povolev15 please confirm

Reply:
Modified and changed to `SHOULD` to reflect that we do not actually enforce the restriction.